src/libraries/System.Linq/src/System/Linq/Distinct.cs

```diff
@@ -20,6 +20,42 @@ public static IEnumerable<TSource> Distinct<TSource>(this IEnumerable<TSource> s
             return new DistinctIterator<TSource>(source, comparer);
         }
 
+        public static IEnumerable<TSource> DistinctBy<TSource, TKey>(this IEnumerable<TSource> source, Func<TSource, TKey> keySelector) => DistinctBy(source, keySelector, null);
```

Comment:
You need to add xml docs for all the new public surface area, right?

Reply:
A few of the PR's from #47231 have already been merged without xml docs. I'm thinking it would make more sense to add everything in one go, as part of the backporting effort in #48986.